gapic/templates/tests/unit/gapic/%name_%version/%sub/test_%service.py.j2

```diff
@@ -1041,9 +1043,17 @@ def test_transport_adc(transport_class):
         transport_class()
         adc.assert_called_once()
 
-{{ test_macros.transport_kind_test(service, opts) }}
-
-{{ test_macros.transport_kind_test(service, opts, is_async=True) }}
+{% set configs = [] %}
+{% for transport_name in opts.transport %}
+    {% do configs.append({'service':service, 'transport_name':transport_name, 'is_async':false}) %}
+{# TODO(#2121): Remove this condition when async rest is GA. #}
+{% if 'grpc' in transport_name or rest_async_io_enabled %}
+    {% do configs.append({'service':service, 'transport_name':transport_name + "_asyncio", 'is_async':true}) %}
```

Comment:
nit: instead of `transport_name` could we have `grpc` or `rest` and then use a macro to get `transport_name` for the given transport?

Reply:
addressed.